modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala

```diff
@@ -69,7 +72,25 @@ object GlobalSnapshotConsensusFunctions {
       val tokenLockEvents = artifact.tokenLockBlocks.map(_.toList.map(TokenLockEvent(_))).getOrElse(List.empty)
       val unpEvents =
         artifact.updateNodeParameters.getOrElse(SortedMap.empty[Id, Signed[UpdateNodeParameters]]).values.map(UpdateNodeParametersEvent(_))
-      val events: Set[GlobalSnapshotEvent] = dagEvents ++ scEvents ++ allowSpendEvents ++ unpEvents ++ tokenLockEvents
+      val cdsEvents = artifact.activeDelegatedStakes
```

Comment:
This is a perfect example of why we should have an abstracted artifact/event type here for all of these new types, so we can avoid this.